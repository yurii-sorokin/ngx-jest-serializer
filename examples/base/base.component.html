<div class="container" #id>
  <div [ngClass]="{'active': !disabled}"></div>
  <ng-container *ngIf="ifPass">
    <input class="no-async" type="text" [value]="promiseValue" />
    <input class="with-async" type="text" [value]="observableValue | async" />
    <input class="2w" type="text" [(value)]="twoWayValue" />
  </ng-container>
  <ng-container *ngIf="ifNotPass">
    <span>Am I visible?</span>
  </ng-container>
  Just text
  {{'yet another text' | uppercase}}
  <button (click)="doStuff()">Click me!</button>
  <svg>
    <rect x="0" y="0" width="100" height="100" />
  </svg>
</div>
