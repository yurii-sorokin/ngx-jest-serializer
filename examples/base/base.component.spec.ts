import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { NO_ERRORS_SCHEMA } from '@angular/core';

import { BaseComponent } from './base.component';

describe('BaseComponent', () => {
  let component: BaseComponent;
  let fixture: ComponentFixture<BaseComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [BaseComponent],
      schemas: [NO_ERRORS_SCHEMA]
    })
      .compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(BaseComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it(`should render something`, () => {
    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });
});
