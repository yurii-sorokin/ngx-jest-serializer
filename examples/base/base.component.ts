import { Observable } from 'rxjs/Observable';
import 'rxjs/add/observable/of';

import { Component, Input, HostBinding } from '@angular/core';

@Component({
  selector: 'xmpl-base',
  templateUrl: './base.component.html'
})
export class BaseComponent {
  @Input() promiseValue = Promise.resolve(1);
  @Input() observableValue = Observable.of(2);
  @Input() twoWayValue = 3;

  @HostBinding('class.bound')
  bound = true;
  @HostBinding('class.not-bound')
  notBound = false;
  @HostBinding('attr.title')
  title = 'super';
  @HostBinding('style.overflow')
  overflow = 'hidden';
  @HostBinding('style.display')
  get display() { return 'block'; }
  @HostBinding('disabled')
  get disabled() { return false; }

  ifPass = true;
  ifNotPass = false;

  doStuff() {}
}
