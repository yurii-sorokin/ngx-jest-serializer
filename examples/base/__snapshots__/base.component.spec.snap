exports[`BaseComponent should render something 1`] = `
<xmpl-base
  class="bound"
  disabled={false}
  style="overflow: hidden; display: block"
  title="super"
>
  <div
    class="container"
  >


    <div
      class="active"
      ng-reflect-ng-class="[object Object]"
    />






    <input
      class="no-async"
      type="text"
      value={
        ZoneAwarePromise {
          "__zone_symbol__state": true,
          "__zone_symbol__value": 1,
        }
      }
    />


    <input
      class="with-async"
      type="text"
      value={2}
    />


    <input
      class="2w"
      type="text"
      value={3}
      valueChange={[Function]}
    />






  Just text
  YET ANOTHER TEXT

    <button
      click={[Function]}
    >
      Click me!
    </button>


    <svg>


      <rect
        height="100"
        width="100"
        x="0"
        y="0"
      />


    </svg>


  </div>


</gp-article-card>
`;
